rpc/filter_api.go

```diff
@@ -0,0 +1,208 @@
+package rpc
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/ethermint/x/evm/types"
+
+	"math/big"
+
+	eth "github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+type FilterCriteria eth.FilterQuery
+
+// PublicFilterAPI is the eth_ prefixed set of APIs in the Web3 JSON-RPC spec.
+type PublicFilterAPI struct {
+	cliCtx context.CLIContext
+}
+
+// NewPublicEthAPI creates an instance of the public ETH Web3 API.
+func NewPublicFilterAPI(cliCtx context.CLIContext) *PublicFilterAPI {
+	return &PublicFilterAPI{
+		cliCtx: cliCtx,
+	}
+}
+
+// GetLogs returns logs matching the given argument that are stored within the state.
+//
+// https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getlogs
+func (e *PublicFilterAPI) GetLogs(criteria FilterCriteria) ([]*ethtypes.Log, error) {
+	var filter *Filter
+	if criteria.BlockHash != nil {
+		/*
+			Still need to add blockhash in prepare function for log entry
+		*/
+		filter = NewBlockFilter(*criteria.BlockHash, criteria.Addresses, criteria.Topics)
+		results := e.getLogs()
+		logs := filterLogs(results, nil, nil, filter.addresses, filter.topics)
+		return logs, nil
+	} else {
+		// Convert the RPC block numbers into internal representations
+		begin := rpc.LatestBlockNumber.Int64()
+		if criteria.FromBlock != nil {
+			begin = criteria.FromBlock.Int64()
+		}
+		from := big.NewInt(begin)
+		end := rpc.LatestBlockNumber.Int64()
+		if criteria.ToBlock != nil {
+			end = criteria.ToBlock.Int64()
+		}
+		to := big.NewInt(end)
+		results := e.getLogs()
```

Comment:
Would it not be beneficial to include the range filter on the type you created to abstract this logic so you aren't doing it duplicate times?

Reply:
can you elaborate on what you mean from the above?

Reply:
Just replicating the e.getLogs and filterLogs calls, can't the filter just be generated in this if/ else then call these equivalently outside of it for consistency?

Reply:
still need some clarity here.... `e.getLogs` is the actual call to retrieve the logs whereas `filterLogs` does the filtering. We want to keep this functionality separate when we introduce bloom filters so that we only call `filterLogs` on the logs that _may_  meet the filter criteria which will be indicated by the bloom filter.

Reply:
to be more specific, I just mean having the functionality to create a range filter like you do with `NewBlockFilter()` then get logs and filter them below this conditional (let me know what you think of this

Reply:
The only problem I have is that you are handling block filters and range filters asymmetrically, which is somewhat hard to follow. Also small thing with this is that it adds a linting issue with returning both in the coniditonal (that doesn't matter as much)